examples/token/nep5/nep5.go

```diff
@@ -35,7 +35,12 @@ func getIntFromDB(ctx storage.Context, key []byte) int {
 
 // GetSupply gets the token totalSupply value from VM storage
 func (t Token) GetSupply(ctx storage.Context) int {
-	return getIntFromDB(ctx, []byte(t.CirculationKey))
+	supply := getIntFromDB(ctx, []byte(t.CirculationKey))
+	if supply == 0 {
+		storage.Put(ctx, []byte(t.CirculationKey), t.TotalSupply)
```

Comment:
get* shouldn't change the DB in any way, it should be possible to run that using read-only snapshot. This Put belongs to minting function actually, every time you mint something you add to `TotalSupply` and save it.